src/vault/matrix/storage.rs

```diff
@@ -0,0 +1,596 @@
+use aes::cipher::generic_array::ArrayLength;
+use aes::cipher::{generic_array::GenericArray, NewCipher, StreamCipher};
+use aes::Aes256Ctr;
+use core::convert::TryInto;
+use hmac::{Hmac, Mac};
+use pbkdf2::pbkdf2;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use sha2::Sha256;
+use sha2::Sha512;
+use std::convert::TryFrom;
```

Comment:
In no_std libraries and for consistency is better to explicitly use `core` or `alloc` unless it's something that only lives in `std`
```suggestion
use core::convert::TryFrom;
```